packages/davinci-client/src/lib/collector.utils.ts

```diff
@@ -341,11 +345,100 @@ export function returnMultiSelectCollector(
 }
 
 /**
- * @function returnMultiValueCollector - Creates a MultiValueCollector object based on the provided field, index, and optional collector type.
+ * @function returnObjectCollector - Creates a ObjectCollector object based on the provided field, index, and optional collector type.
  * @param {DaVinciField} field - The field object containing key, label, type, and links.
- * @param {number} idx - The index to be used in the id of the MultiValueCollector.
- * @param {MultiValueCollectorTypes} [collectorType] - Optional type of the MultiValueCollector.
- * @returns {MultiValueCollector} The constructed MultiValueCollector object.
+ * @param {number} idx - The index to be used in the id of the ObjectCollector.
+ * @param {ObjectValueCollectorTypes} [collectorType] - Optional type of the ObjectCollector.
+ * @returns {ObjectCollector} The constructed ObjectCollector object.
+ */
+export function returnObjectCollector<
+  Field extends DeviceAuthenticationFieldValue | DeviceRegistrationFieldValue,
+  CollectorType extends ObjectValueCollectorTypes = 'ObjectValueCollector',
+>(field: Field, idx: number, collectorType: CollectorType) {
```

Comment:
I'm wondering if we should restructure how across the board we are doing function arguments to be composable.

I just think how can we use these functions within a `pipe` and it's pretty difficult.

Reply:
Yeah, I agree. As to not make this into a large refactor, let's save these changes until the implementation of our new "pipe" solution.